client/js/ifml/tojson/elementrules.js

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2018, the IFMLEdit.org project authors. Please see the
+// AUTHORS file for details. All rights reserved. Use of this source code is
+// governed by a MIT-style license that can be found in the LICENSE file.
+/*jslint node: true, nomen: true */
+"use strict";
+
+var almost = require('almost'),
+    createRule = almost.createRule;
+
+module.exports = [
+    createRule(
+        function (element) {
+            return ['ifml.Event', 'ifml.Action', 'ifml.ViewComponent', 'ifml.ViewContainer'].includes(element.get('type'));
```

Comment:
You sould still be able to use _almost-extend_ and define custom helpers for this checks.
https://github.com/B3rn475/almostjs-extend#custom-helpers